AWSAppSyncClient/AWSAppSyncClient.swift

```diff
@@ -184,9 +183,12 @@ public class AWSAppSyncClient {
         } catch {
```

Comment:
Need to validate if Promise resolved to failure

Reply:
await should throw the error if failure